### ISSUE TITLE
Web - Draft was added to ~20 chats without any action taken by the user

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#26680]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. User walked away from their computer for 10 minutes
2. They returned and refreshed NewDot
3. They noticed about 20 chats that showed the draft pencil icon
4. They opened each chat and the text box said `iH`.

## Expected Result:
Drafts should not be added to chats unless the user starts typing a message in that chat.

## Actual Result:
Duplicate drafts were added to multiple chats without any action by the user.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.58.5
**Reproducible in staging?:** n/a
**Reproducible in production?:** n/a
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

![image (29)](https://github.com/Expensify/App/assets/93399543/fd41c851-0d4a-4ded-8a7e-3cc91e28a8e3)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @sakluger 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1693447325495479

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by sophiepintoraetz** [comment]
This happened to me as well -

I left my computer ~11;50am NZT
Returned to my computer ~13:05 NZT
found that some of my chats had the draft pencil icon and the messaging `m'l`
(I don't know why some chats had the draft and not others - as it affected some of my pinned chats but not others, one of my unread messages but not others)
![image](https://github.com/Expensify/App/assets/42940078/6891824b-22da-4883-805e-7d476b7491b1)

**Comment by jliexpensify** [comment]
Interesting, thanks @sophiepintoraetz ! 

Hmm I wonder if this might be related to the speed of typing in New.Dot? Sometimes when I'm typing pretty quickly, I notice New.Dot has a lag, and then when it "catches up", I notice jumbled words that show as drafts - almost like New.Dot has caught up with the typing, but since you've already sent the message, it has these extra excess letters.

I probably sound crazy, but I definitely think there's a huge lag in New.Dot atm. Making this External!

**Comment by sophiepintoraetz** [comment]
I don't see how it could be related though - I hadn't typed anything at the time, just simply opened up my newdot and observed that there were drafts.

**Comment by RachCHopkins** [comment]
I also just had this happen! I was showing Ben how to use newdot about an hour ago, I just went back to newdot and I have three drafts. Two say `ram` and one says `rarm`.

<img width="363" alt="image" src="https://github.com/Expensify/App/assets/32854888/da39e38c-d440-476f-bd1d-b752694823fd">

<img width="533" alt="image" src="https://github.com/Expensify/App/assets/32854888/870359e4-348b-4b58-9fc6-125bcfe6aad1">

<img width="280" alt="image" src="https://github.com/Expensify/App/assets/32854888/5de65679-c5a1-4b73-a007-f4a9406a7363">

<img width="537" alt="image" src="https://github.com/Expensify/App/assets/32854888/779c8115-b566-490e-965e-f8b77c9b7142">

I have absolutely no idea where or why I would have been typing `ram` or `rarm`

**Comment by sakluger** [comment]
I think this is related to the app lagging while typing. I tried typing `lol damn` and then after some serious lag, it output the following:

<img width="1194" alt="image" src="https://github.com/Expensify/App/assets/5997400/7814ec1f-784c-44b2-b685-1025575722d1">

**Comment by mountiny** [comment]
Also a similar reports:
- https://expensify.slack.com/archives/C049HHMV9SM/p1694119295980249 cc @danielrvidal 
- https://expensify.slack.com/archives/C049HHMV9SM/p1694130035779309 cc @cead22

**Comment by zukilover** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Drafts should not be added to chats unless the user starts typing a message in that chat.

This bug can simply be reproduced by:

1. Clicking on a workspace, then click on the workspace header (to trigger a blur event for the chat composer)
2. Start typing on your keyboard (without focusing on any input fields)
3. Click outside the workspace detail to close it, then click anywhere in the workspace outside the chat composer
4. Again, start typing on your keyboard without focusing on any input fields
5. Click on another workspace, and repeat step 1 - 4

### What is the root cause of that problem?
`replaceSelectionWithText` that is responsible for text insertion at `selection` point is excessively called within `focusComposerOnKeyPress` which is a handler for various event including navigation blur/focus. This is unnecessary because `replaceSelectionWithText` is already being executed in an `useImperativeHandle` hook.

https://github.com/Expensify/App/blob/fc8adf99a11a41154ec276d44495485958e43a7d/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js#L387-L390

### What changes do you think we should make in order to solve the problem?
Remove `replaceSelectionWithText` from `focusComposerOnKeyPress` event handler, on line 388:
```
                   focus();
            // Remove this: replaceSelectionWithText(e.key, false);
        },
        [
            checkComposerVisibility, 
            focus, 
            // Remove this: replaceSelectionWithText
        ],
```

### What alternative solutions did you explore? (Optional)
N/A

**Comment by b4s36t4** [comment]
@mountiny For the moment https://github.com/Expensify/App/pull/26826 this PR solves the issue. Although RCA is different but ultimately we should never focus on the composer if the route is not focused. I have tested this bug report https://expensify.slack.com/archives/C049HHMV9SM/p1693995458316799 with the above PR seems it's not re-produceable.

**Comment by jliexpensify** [comment]
Bumo @mananjadhav to check out the above PR please!

**Comment by mountiny** [comment]
Put it on hold for your Pr and we can see if it will be fixed @b4s36t4

**Comment by jliexpensify** [comment]
Changing the labels since it's on hold atm

**Comment by jliexpensify** [comment]
https://github.com/Expensify/App/pull/26826 has been deployed last month. I don't know how to repro this - it seems to happen randomly to some people. It hasn't happened to me in the last month.

Should we close this @mountiny ?

**Comment by mountiny** [comment]
I dont think this was reproed since then really so yeah I think we can close this
